test/cppsuite/test_harness/workload_generator.h

```diff
@@ -233,37 +238,46 @@ class workload_generator : public component {
         wt_timestamp_t ts;
         std::vector<WT_CURSOR *> cursors;
         std::string collection_name;
-        std::vector<std::string> collection_names;
+        std::vector<std::string> collection_names = context.get_collection_names();
+        std::map<std::string, std::vector<key_value_t>> collection_keys;
         key_value_t generated_value, key;
         bool has_committed = true;
         int64_t cpt, value_size = context.get_value_size();
 
         testutil_assert(session != nullptr);
         /* Get a cursor for each collection in collection_names. */
-        for (const auto &it : context.get_collection_names()) {
+        for (const auto &it : collection_names) {
             testutil_check(session->open_cursor(session, it.c_str(), NULL, NULL, &cursor));
             cursors.push_back(cursor);
-            collection_names.push_back(it);
+            collection_keys[it] = context.get_collection_keys(it);
         }
 
-        while (context.is_running()) {
-            /* Walk each cursor. */
-            context.begin_transaction(session, "");
-            ts = context.set_commit_timestamp(session);
-            cpt = 0;
-            /* The key to update is hard coded to 1 for now. */
-            key = 1;
-            for (const auto &it : cursors) {
-                collection_name = collection_names[cpt];
+        cpt = 0;
+        /* Walk each cursor. */
+        for (const auto &it : cursors) {
+            collection_name = collection_names[cpt];
+            /* Walk each key. */
+            for (auto const &key : collection_keys.at(collection_name)) {
+                if (has_committed) {
```

Comment:
Rather than returning a bool from  commit_transaction can we have a context.in_txn() which returns a bool?